Patterns/DependencyInjection.md

```diff
@@ -128,19 +128,40 @@ services.AddTransient(typeof(IRepository<>), typeof(Repository<>));
 
 ### Registering `internal` services
 
-When building class libraries it is a good practice to make library specific details invisible by marking some classes as `internal`. A service in `Contoso.Services` just needs `IRepository<SomeEntity>` from `Contoso.Repositories` and should not be able to see or touch `SomeEntityRepositoryImplementation`. But when registering that repository in the DI container `Contoso.API` a developer can no longer do `services.AddScoped<IRepository<SomeEntity>, SomeEntityRepositoryImplementation>()` as `Contoso.API` cannot access the internals of `Contoso.Repositories`. To solve this, every class library should have a `DependencyConfiguration.cs` file at the root of its project, containing something like this:
+When building class libraries it is a good practice to make library specific details invisible by marking some classes as `internal`. A service in `Contoso.Services` just needs `IRepository<SomeEntity>` from `Contoso.Repositories` and should not be able to see or touch `SomeEntityRepositoryImplementation`. But when registering that repository in the DI container `Contoso.API` a developer can no longer do `services.AddScoped<IRepository<SomeEntity>, SomeEntityRepositoryImplementation>()` as `Contoso.API` cannot access the internals of `Contoso.Repositories`. 
+To solve this, every class library should provide extension method on IServiceCollection like this:
 
 ```c#
 namespace Contoso.Repositories
 {
-    public static class DependencyConfiguration 
+    public static class ContosoRepositoriesServiceCollectionExtensions 
     {
-        public static void Register(IServiceCollection services) 
+        public static void AddContosoRepositories(this IServiceCollection services) 
         {
             services.AddScoped<IRepository<SomeEntity>, SomeEntityRepositoryImplementation>();
         }
     }
 }
 ```
 
-Because this class is in the `Repositories` project, it can reference the internal `SomeEntityRepositoryImplementation`. `Contoso.API` can simply call `Repositories.DependencyConfiguration.Register(services)` and have all the appropriate services registered. Keep in mind that this approach is all or nothing, you either register all the services of a project, or none. If that is problematic, either split the project into multiple project as it has become too broad and contains a diffuse set of classes or split the `Register` method into multiple methods and call one or multiple of them in projects that use the library.
+Because this class is in the `Repositories` project, it can reference the internal `SomeEntityRepositoryImplementation`. `Contoso.API` can simply call `services.AddContosoRepositories()` and have all the appropriate services registered. Keep in mind that this approach should be all or nothing, you either take full control of dependency registration in your library, or leave it entirely up to users of your library what services to inject. 
```

Comment:
Perhaps also mention that a `Contoso.Services` project is also allowed to call `.AddContosoRepositories`.